File: backend/src/main/java/com/koval/trainingplannerbackend/club/ClubTrainingSession.java
package com.koval.trainingplannerbackend.club;

import lombok.Getter;
import lombok.Setter;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;
import java.util.Objects;

@Getter
@Setter
@Document(collection = "club_training_sessions")
public class ClubTrainingSession {
    @Id
    private String id;

    @Indexed
    private String clubId;

    private String createdBy;
    private String title;
    private String sport;
    private LocalDateTime scheduledAt;
    private String location;
    private String description;
    private String linkedTrainingId;
    private List<String> participantIds = new ArrayList<>();
    private LocalDateTime createdAt;
    private String recurringTemplateId;
    private Integer maxParticipants;
    private List<WaitingListEntry> waitingList = new ArrayList<>();

    public boolean isFull() {
        return maxParticipants != null && participantIds.size() >= maxParticipants;
    }

    public boolean isOnWaitingList(String userId) {
        return waitingList.stream().anyMatch(e -> Objects.equals(e.userId(), userId));
    }
}

File: android/app/src/main/java/com/koval/trainingplanner/domain/model/ClubTrainingSession.kt
package com.koval.trainingplanner.domain.model

data class ClubTrainingSession(
    val id: String,
    val clubId: String,
    val title: String,
    val sport: SportType? = null,
    val scheduledAt: String, // ISO datetime
    val location: String? = null,
    val description: String? = null,
    val participantIds: List<String> = emptyList(),
    val waitingList: List<String> = emptyList(),
    val maxParticipants: Int? = null,
    val durationMinutes: Int? = null,
    val linkedTrainings: List<LinkedTraining> = emptyList(),
    val clubGroupId: String? = null,
    val clubGroupName: String? = null,
    val responsibleCoachId: String? = null,
    val responsibleCoachName: String? = null,
    val cancelled: Boolean = false,
    val cancellationReason: String? = null,
    val clubName: String? = null,
)

data class LinkedTraining(
    val trainingId: String,
    val title: String? = null,
    val clubGroupId: String? = null,
)
